### PR TITLE
DDF-4179 Map pan & zoom by geographic place name appears to search forever and does not reorient the camera

### PR DESCRIPTION
**Pull request by Lambeaux**
#### What does this PR do?
Fixes the pan & zoom capabilities of both maps. 

#### Who is reviewing it? 
@Variadicism 	
@samuelechu 	

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@djblue 
@millerw8

#### How should this be tested?
Using both the 3D and 2D maps, type various queries into the pan / zoom search box, such as "par", "tempe", etc and ensure either hitting ENTER or picking from autocomplete moves the camera to the intended target. 

#### What are the relevant tickets?
[DDF-4179](https://codice.atlassian.net/browse/DDF-4179)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js [hunk 1]
```diff
@@ -66,38 +66,44 @@ module.exports = Backbone.Model.extend({
...
+        data: 'jsonp=jsonp&query=' + input,
+        contentType: 'application/javascript',
+        dataType: 'jsonp',
+        jsonp: 'jsonp',
+        success: function (result) {  
+          // Extract desired fields
+          result = [result]
+          resolve(result.map(function (locationResult) {
+            var resultValues = locationResult.resourceSets[0].resources[0]
+            if (typeof resultValues != 'undefined') {
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: a ternary might be more readable here

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js [hunk 2]
```diff
@@ -66,38 +66,44 @@ module.exports = Backbone.Model.extend({
...
+    return new Promise((resolve, reject) => {
+      $.ajax({
+        url: geocoderOfflineEndpoint,
+        data: 'jsonp=jsonp&query=' + input,
+        contentType: 'application/javascript',
+        dataType: 'jsonp',
+        jsonp: 'jsonp',
+        success: function (result) {  
+          // Extract desired fields
+          result = [result]
```
Comment (dimitry-sherman) [inline ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: :pencil2: Being immutable whenever possible might be a good pattern to follow
```javascript
const resultValues = result.resourceSets[0].resources[0]
```

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js (hunk 1 above)
Comment (dimitry-sherman) [reply ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: :exclamation: Equality is usually more readable than in-equality, and **it should be `!==` or `===`**.

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js [hunk 3]
```diff
@@ -39,11 +39,11 @@ module.exports = Backbone.Model.extend({
   queryGazetteerOnline(input) {
     // Load in request string
     return Cesium.loadJson(geocoderOnlineEndpoint + input)
-      .then(function(results) {
+      .then(function (results) {
```
[the remark this answers is not in the document]
Comment (Lambeaux) [reply ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: yes - fixed

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js (hunk 2 above)
Comment (BernardIgiri) [reply ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: I would add that you should use `let` and `const` to replace all instances of `var` with `const` being preferred.

**Comment by BernardIgiri** [comment]
This needs a hero. Dimitry is on vacation so we can skip his approval.

**Comment by Lambeaux** [comment]
Also fixed the online gazetteer. 
Build now.

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js [hunk 4]
```diff
@@ -38,66 +37,72 @@ module.exports = Backbone.Model.extend({
...
+      $.ajax({
+        url: geocoderOnlineEndpoint + input,
+        success: function(results) {
+          const formattedResults = results.map(function(resultObject) {
+            const bboxDegrees = resultObject.boundingbox
             return {
-              displayName: resultValues.name,
+              displayName: resultObject.display_name,
               destination: Cesium.Rectangle.fromDegrees(
-                resultValues.bbox[1],
```
Comment (adimka) [inline ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: ❓ Is the Bounding Box coordinate order different for the online vs offline gazetteer?

Comment (Lambeaux) [reply ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js]: This is how it has always been. See master: 
https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/geocoder.js#L51-L85

**Pull request merged**